common/src/bitset.rs

```diff
@@ -198,16 +301,160 @@ impl BitSet {
     }
 }
 
+/// Lazy Read a serialized BitSet.
+#[derive(Clone)]
+pub struct ReadSerializedBitSet {
+    data: OwnedBytes,
+    max_value: u32,
+}
+
+impl ReadSerializedBitSet {
+    pub fn new(data: OwnedBytes) -> Self {
+        let (max_value_data, data) = data.split(4);
+        let max_value: u32 = u32::from_le_bytes(max_value_data.as_ref().try_into().unwrap());
+        ReadSerializedBitSet { data, max_value }
+    }
+
+    /// Count the number of unset bits from serialized data.
+    ///
+    #[inline]
+    pub fn count_unset(&self) -> usize {
+        let lower = self.max_value % 64u32;
+
+        let num_set: usize = self
+            .iter_tinysets()
+            .map(|(tinyset, is_last)| {
+                if is_last {
+                    tinyset.intersect(TinySet::range_lower(lower)).len() as usize
+                } else {
+                    tinyset.len() as usize
+                }
+            })
+            .sum();
+        self.max_value as usize - num_set
+    }
+
+    /// Iterate the tinyset on the fly from serialized data.
+    ///
+    /// Iterator returns (TinySet, is_last) element, so the consumer can ignore up to max_doc in the
+    /// last block.
+    ///
+    #[inline]
+    fn iter_tinysets<'a>(&'a self) -> impl Iterator<Item = (TinySet, bool)> + 'a {
```

Comment:
If it reaches this level of complexity, I think we'd better access the remaining bits but make sure they are set to their right value.

We can do that by either reading the actually bits or masking the like TinySet.
I would rather not care, and read the actual bits and fix the initialization `with_max_value_and_full`